a3fe/tests/test_engine_configuration.py

```diff
@@ -0,0 +1,245 @@
+            input_dir=input_dir
+        )
+        assert isinstance(config, SomdConfig)
+        
+        # Test with custom stream_log_level
+        config = SomdConfig(
+            runtime=1,
+            input_dir=input_dir,
+            stream_log_level=logging.DEBUG
+        )
+        assert config._stream_log_level == logging.DEBUG
+
+def test_config_yaml_save_and_load():
+    """Test that the config can be saved to and loaded from YAML."""
+    with create_test_input_dir() as input_dir:
+        with TemporaryDirectory() as dirname:
+            config = SomdConfig(
+                runtime=1,  # Integer runtime
+                input_dir=input_dir
+            )
+            yaml_path = os.path.join(dirname, "config.yaml")
+            config.dump(yaml_path)
+            config2 = SomdConfig.load(yaml_path)
+            assert config == config2
+
+def test_get_somd_config():
+    """Test that the SOMD configuration file is generated correctly."""
+    with create_test_input_dir() as input_dir:
+        with TemporaryDirectory() as dirname:
+            config = SomdConfig(
+                integrator="langevinmiddle",
+                nmoves=25000,
+                timestep=4.0,
+                runtime=1,  # Integer runtime
+                input_dir=input_dir,
+                cutoff_type="PME",
+                thermostat=False
+            )
+            config_path = config.get_somd_config(
+                run_dir=dirname,
+                config_name="test"
+            )
+            assert config_path == os.path.join(dirname, "test.cfg")
+
+            expected_config = (
+                "### Integrator ###\n"
+                "nmoves = 25000\n"
+                "timestep = 4.0 * femtosecond\n"
+                "constraint = hbonds\n"
+                "hydrogen mass repartitioning factor = 3.0\n"
+                "integrator = langevinmiddle\n"
+                "inverse friction = 1.0 * picosecond\n"
+                "temperature = 25.0 * celsius\n"
+                "thermostat = False\n"
+                "\n"
+                "### Barostat ###\n"
+                "barostat = True\n"
+                "pressure = 1.0 * atm\n"
+                "\n"
+                "### Non-Bonded Interactions ###\n"
+                "cutoff type = PME\n"
+                "cutoff distance = 10.0 * angstrom\n"
+                "\n"
+                "### Trajectory ###\n"
+                "buffered coordinates frequency = 5000\n"
+                "center solute = True\n"
+                "\n"
+                "### Minimisation ###\n"
+                "minimise = True\n"
+                "\n"
+                "### Alchemistry ###\n"
+                "perturbed residue number = 1\n"
+                "energy frequency = 200\n"
+            )
+
+            with open(config_path, "r") as f:
+                config_content = f.read()
+
+            assert config_content == expected_config
+
+def test_constraint_validation():
```

Comment:
Please look into useing pytest.parameterize to rerun the same test with different arguments (and results).